### PR TITLE
feat: named view support

### PR DESCRIPTION
**Pull request by hannoeru**

### CODE REVIEW

File: src/core/treeLeafValue.ts [hunk 1]
```diff
@@ -34,7 +34,9 @@ class _TreeLeafValueBase {
   /**
    * Does the node has a component at the given path. e.g having `routes/users/index.vue` and `routes/users.vue`.
    */
-  filePath?: string
+  filePaths?: {
```
Comment (posva) [inline src/core/treeLeafValue.ts:37]: I think we should just use a `Map<string, string>`. It should be easier to check if it's empty or not

File: e2e/__snapshots__/routes.spec.ts.snap [hunk 2]
```diff
@@ -66,13 +82,17 @@ exports[`generates the routes 1`] = `
       {
         path: \\":doc?\\",
```
Comment (posva) [inline e2e/__snapshots__/routes.spec.ts.snap:83]: I think we should use single quotes to simplify the snapshots reading 😆 . I will change that later on

File: playground/src/App.vue [hunk 3]
```diff
@@ -22,5 +22,7 @@ route.name === '/about'
     </div>
   </header>
 
-  <!-- <RouterView /> -->
+  <RouterView />
+  <hr />
+  <RouterView name="named" />
```
Comment (posva) [inline playground/src/App.vue:27]: Later on we could also type this 😎

**Comment by posva** [comment]
I fixed the errors on windows, you can rebase @hannoeru

File: src/core/context.ts [hunk 4]
```diff
@@ -63,8 +63,10 @@ export function createRoutesContext(options: Required<Options>) {
 
   function addPage(path: string) {
     log('added', path)
+    const route = stripRouteFolder(path)
+    // TODO: handle top level named view HMR
```
Comment (posva) [inline src/core/context.ts:67]: what did you mean with this?

**Review by posva — APPROVED** [review]
Thanks a lot! I just found a few bugs I introduced myself early on and I going to refactor a little bit the splitting of the path after merging

File: src/core/context.ts (hunk 4 above)
Comment (hannoeru) [reply src/core/context.ts:67]: I haven't dig into it yet, but currently named view has some bug with HMR when you change routes's filename.

**Pull request merged**